server/handler/validate.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+
+	"github.com/palantir/go-baseapp/baseapp"
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v2"
+
+	"github.com/palantir/policy-bot/policy"
+	"github.com/palantir/policy-bot/version"
+)
+
+type ValidateCheck struct {
+	Message string `json:"message"`
+	Version string `json:"version"`
+}
+
+func Validate() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := zerolog.Ctx(ctx)
+
+		logger.Info().Msg("Attempting to validate policy file")
+		file, _, err := r.FormFile("policy")
```

Comment:
It might be better to PUT/POST the content of the file as the request body (and then read it from `r.Body` here), rather than use the form file upload functionality. Maybe there's no difference in practice, but form file handling has always felt more complicated to me, particularly when using `curl` to make requests.

Reply:
Initially I had issues using `curl -d`, and used the form uploads. I realized that my issue was with the extra processing curl does, and I needed to specify `--data-binary`. I've also noted this in the README, but it is a bit of a frog trap.

Reply:
Do we have the same problem if we use `PUT` instead of `POST` and recommend using `-T` / `--upload-file` with `curl`?

Reply:
There doesn't seem to be an issue using PUTs and it cleans up the usage, so lets go with that instead.